### DISCUSSION TITLE
Is it possible to stream files to clients without causing a client-sided block?

### DISCUSSION THREAD

**Discussion by julianbibo** [miguelgrinberg/python-socketio#733] [Q&A]
I'm trying to send medium-sized audio files over socketio, except the JS client hangs while it is receiving the emitted data, and no other packets can be received during that time. Is it possible to stream files to clients without causing the client's socketio.on(..) handler to block everything?

I found an implementation of socket.io-stream here, also wondering if there is a Python implementation of this: https://stackoverflow.com/a/40391620/13216113

Love your work Miguel!

**Comment by miguelgrinberg** [comment]
Isn't this a question that needs to be asked to the maintainers of the JS client? I can't really answer for a problem in the JS client.

**Comment by julianbibo** [comment]
I don't know the specific workings of python-socketio, but does it support streaming at all? Say, if the JS client can support streams (socket.io-stream), can it receive "normal" non-stream emits from the python-socketio server?

**Comment by miguelgrinberg** [comment]
The Socket.IO protocol doesn't have anything called "streaming". I think you are looking at an extension built by a 3rd party that works on top of the standard Socket.IO packet exchange mechanisms.

**Comment by julianbibo** [comment]
Alright thanks! I'll look into it
